sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -431,17 +431,37 @@ trait CheckAnalysis extends PredicateHelper {
               // include collections because structs nested in maps and arrays may be altered
               val field = table.schema.findNestedField(fieldName, includeCollections = true)
               if (field.isEmpty) {
-                throw new AnalysisException(
-                  s"Cannot $operation missing field in ${table.name} schema: ${fieldName.quoted}")
+                alter.failAnalysis(
+                  s"Cannot $operation missing field ${fieldName.quoted} in ${table.name} schema: " +
+                  table.schema.treeString)
+              }
+              field.get._2
+            }
+            def positionArgumentExists(position: ColumnPosition, dataType: DataType): Boolean = {
+              (position, dataType) match {
+                case (after: After, struct: StructType) =>
+                  struct.fieldNames.contains(after.column())
+                case (after: After, _) => false
```

Comment:
shall we fail for this case? We can't add fields to a non-struct column.

Reply:
Since this is used for non add-column methods as well, I left it a bit more generic for now